codelab/JinghangCodeLab/MagicNumberConverter.java

```diff
@@ -0,0 +1,41 @@
+package JinghangCodeLab;
+
+public class MagicNumberConverter {
+
+    public static int binaryToDecimal(String binary)
+    {  
+        //edge cases
+        if (binary == "0"){
+            return 0;
+        }
+
+        //negative
+        int isNegative = -1;
```

Comment:
isNegative variable can be a boolean type instead of "int" type. making it "boolean" type will improve code's readability.